Source/WebCore/workers/service/WorkerSWClientConnection.cpp

```diff
@@ -551,4 +557,64 @@ void WorkerSWClientConnection::retrieveRecordResponseBody(BackgroundFetchRecordI
     });
 }
 
+void WorkerSWClientConnection::addCookieChangeSubscriptions(ServiceWorkerRegistrationIdentifier registrationIdentifier, Vector<CookieChangeSubscription>&& subscriptions, ExceptionOrVoidCallback&& callback)
+{
+    auto requestIdentifier = SWClientRequestIdentifier::generate();
+    m_voidCallbacks.add(requestIdentifier, WTFMove(callback));
+
+    callOnMainThread([thread = m_thread, requestIdentifier, registrationIdentifier, subscriptions = crossThreadCopy(WTFMove(subscriptions))]() mutable {
+        Ref connection = ServiceWorkerProvider::singleton().serviceWorkerConnection();
+        connection->addCookieChangeSubscriptions(registrationIdentifier, WTFMove(subscriptions), [thread = WTFMove(thread), requestIdentifier](auto&& result) {
+            thread->runLoop().postTaskForMode([requestIdentifier, result = crossThreadCopy(WTFMove(result))](auto& scope) mutable {
+                auto callback = downcast<WorkerGlobalScope>(scope).swClientConnection().m_voidCallbacks.take(requestIdentifier);
+                if (!callback) {
+                    callback(Exception { ExceptionCode::SecurityError, "Missing callback"_s });
```

Comment:
How is this a SecurityError? I recommend `InvalidStateError`. Also, callback is internal WebKit state so "Missing callback" is not going to be useful to the Web developer getting this error.